goal_src/jak1/engine/gfx/font.gc

```diff
@@ -1830,3 +1830,40 @@
     (draw-string str buf context)
     )
   )
+
+;; Added for PC port
+(defun draw-string-xy-scaled ((str string) (buf dma-buffer) (x int) (y int) (color font-color) (flags font-flags) (scale float))
+  "Draw a string at the given xy location, with the given scale."
+  (let ((font-ctxt (new 'stack 'font-context *font-default-matrix* x y 0.0 color flags)))
+    (let ((scaled-width (* (-> font-ctxt width) scale))
+          (scaled-height (* (-> font-ctxt height) scale))
+          (orig-mat-x-scale (-> font-ctxt mat vector 0 x))
+          (orig-mat-y-scale (-> font-ctxt mat vector 1 y))
+          (orig-vid-parms-x-scale (-> *video-parms* relative-x-scale))
+          (orig-vid-parms-y-scale (-> *video-parms* relative-y-scale))
+          )
+      (*! (-> font-ctxt scale) scale)
+      (set! (-> font-ctxt width) scaled-width)
+      (set! (-> font-ctxt height) scaled-height)
+      
+      ;; scaling globals *font-default-matrix*, *video-parms* (based on print-game-text scaling)
+      (*! (-> *font-default-matrix* vector 0 x) scale)
+      (*! (-> *font-default-matrix* vector 1 y) scale)
+      (*! (-> *video-parms* relative-x-scale) scale)
+      (*! (-> *video-parms* relative-y-scale) scale)
+      (set! (-> *video-parms* relative-x-scale-reciprical) (/ 1.0 (-> *video-parms* relative-x-scale)))
+      (set! (-> *video-parms* relative-y-scale-reciprical) (/ 1.0 scale))
+
+      (draw-string str buf font-ctxt)
+      
+      ;; un-scaling globals *font-default-matrix*, *video-params*
+      (set! (-> *font-default-matrix* vector 0 x) orig-mat-x-scale)
+      (set! (-> *font-default-matrix* vector 1 y) orig-mat-y-scale)
+      (set! (-> *video-parms* relative-x-scale) orig-vid-parms-x-scale)
+      (set! (-> *video-parms* relative-y-scale) orig-vid-parms-y-scale)
+      (set! (-> *video-parms* relative-x-scale-reciprical) (/ 1.0 (-> *video-parms* relative-x-scale)))
+      (set! (-> *video-parms* relative-y-scale-reciprical) (/ 1.0 (-> *video-parms* relative-y-scale)))
+      )
+    )
```

Comment:
Can you not just do
```lisp
  (let ((context (new 'stack 'font-context *font-default-matrix* x y 0.0 color flags)))
    (set-scale! context scale)
    (draw-string str buf context)
    )
```
?

Reply:
No, `draw-string` doesn't actually respect the `font-context` scale, as far as I could tell. `print-game-text-scaled` needed to take that scale and apply it to the matrix and video-parms, and then revert them back after.

Reply:
I could move this scaling logic into draw-string I suppose, but that function is a mess and I was reaaaaally reluctant to touch it

Reply:
Oh okay, it's fine then. Don't touch that function, it's a tangled mess of assembly.